jbmc/src/java_bytecode/java_trace_validation.cpp

```diff
@@ -115,6 +115,18 @@ bool check_address_structure(const address_of_exprt &address)
   return symbol_expr && check_symbol_structure(*symbol_expr);
 }
 
+bool check_constant_structure(const constant_exprt &constant_expr)
+{
+  if(constant_expr.has_operands())
+  {
+    const auto &operand = skip_typecast(constant_expr.operands()[0]);
+    return operand.id() == ID_constant || operand.id() == ID_address_of ||
+           operand.id() == ID_plus;
```

Comment:
⛏ I don't mind, but people will complain this isn't done with `can_cast_expr<plus_exprt>`, etc.